autoblocks/_impl/prompts/autogenerate.py

```diff
@@ -0,0 +1,327 @@
+import re
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Set
+from typing import Tuple
+
+import httpx
+
+from autoblocks._impl.config.constants import API_ENDPOINT
+from autoblocks._impl.prompts.constants import LATEST
+from autoblocks._impl.prompts.models import AutogeneratePromptsConfig
+from autoblocks._impl.prompts.models import FrozenModel
+from autoblocks._impl.util import AutoblocksEnvVar
+
+
+class TemplateParam(FrozenModel):
+    name: str
+
+    @property
+    def snake_case_name(self) -> str:
+        return to_snake_case(self.name)
+
+
+class Template(FrozenModel):
+    id: str
+    params: List[TemplateParam]
+
+    @property
+    def snake_case_id(self) -> str:
+        return to_snake_case(self.id)
+
+
+class Prompt(FrozenModel):
+    id: str
+    major_version: str
+    has_multiple_major_versions: bool
+    params: Dict[str, Any]
```

Comment:
does params get set to an empty dictionary if it's not defined in the ui?

Reply:
yeah i'm setting it to an empty dictionary when parsing the API response:

```python
        try:
            params = row["params"]["params"] or {}
        except KeyError:
            params = {}
```

could also make this `Optional[Dict[str, Any]]` but has the same effect either way in the code that's produced, which will just be an empty pydantic class:

```python
    auto = f"class {prompt.params_class_name}(FrozenModel):\n"

    if not prompt.params:
        auto += f"{indent()}pass\n"
        return auto
```

Reply:
there is an example in the test here as well: https://github.com/autoblocksai/python-sdk/pull/88/files#diff-2575c7a671ecad6ae75e68531604c281fc160335aa67056ff0b839eeb4908d23R361-R362

Reply:
potentially a better option here though is to make the `params` property on the execution context have a None type if there are no params, would probably lead to better typing hinting rather than an empty pydantic class. let me mess with it a bit

Reply:
hmm yeah in pycharm at least it doesn't seem to complain if you reference an attribute that doesn't exist on an empty class, whereas it will if you try to reference an attribute on None

<img width="393" alt="Screenshot 2024-01-16 at 1 19 48 PM" src="https://github.com/autoblocksai/python-sdk/assets/7498009/b4279dd7-60fe-43a6-91ff-16e511cc6a5f">